.github/workflows/test-and-build.yml

```diff
@@ -0,0 +1,21 @@
+name: Test and build
+on:
+  push:
+    branches:
+      - 'main'
```

Comment:
Don't we want to run the build on all branches?

Reply:
Not sure. It it's not a PR for `main`, and not `main` what would be the value of the build?